motor/discovery/k8s/resolver_namespace.go

```diff
@@ -0,0 +1,123 @@
+package k8s
+
+import (
+	"context"
+	"strings"
+
+	"github.com/gobwas/glob"
+	"github.com/rs/zerolog/log"
+	"go.mondoo.com/cnquery"
+	"go.mondoo.com/cnquery/motor/asset"
+	"go.mondoo.com/cnquery/motor/discovery/common"
+	"go.mondoo.com/cnquery/motor/providers"
+	"go.mondoo.com/cnquery/motor/providers/k8s"
+	"go.mondoo.com/cnquery/motor/providers/k8s/resources"
+	"go.mondoo.com/cnquery/motor/vault"
+	"k8s.io/apimachinery/pkg/api/errors"
+)
+
+var _ common.ContextInitializer = (*NamespaceResolver)(nil)
+
+type NamespaceResolver struct{}
+
+func (r *NamespaceResolver) Name() string {
+	return "Kubernetes Namespace Resolver"
+}
+
+func (r *NamespaceResolver) AvailableDiscoveryTargets() []string {
+	return []string{
+		common.DiscoveryAuto,
+		common.DiscoveryAll,
+		DiscoveryClusters,
+		DiscoveryPods,
+		DiscoveryJobs,
+		DiscoveryCronJobs,
+		DiscoveryStatefulSets,
+		DiscoveryDeployments,
+		DiscoveryReplicaSets,
+		DiscoveryDaemonSets,
+		DiscoveryContainerImages,
+		DiscoveryAdmissionReviews,
```

Comment:
I currently suspect this would break the admission webhook: https://github.com/mondoohq/mondoo-operator/blob/main/cmd/mondoo-operator/webhook/cmd.go#L80

I'm not sure how to test this.

Reply:
I managed to test it with a local minkube. Works as you expected. Nice!

Webhook filters objects before sending them to the scan API when they match the namespaces filter.